sapi/phpdbg/phpdbg_info.c

```diff
@@ -403,12 +403,16 @@ PHPDBG_INFO(classes) /* {{{ */
 		phpdbg_print_class_name(ce);
 
 		if (ce->parent) {
-			zend_class_entry *pce;
-			pce = ce->parent;
-			do {
-				phpdbg_out("|-------- ");
-				phpdbg_print_class_name(pce);
-			} while ((pce = pce->parent));
+			if (ce->ce_flags & ZEND_ACC_LINKED) {
+				zend_class_entry *pce;
+				pce = ce->parent;
+				do {
+					phpdbg_out("|-------- ");
+					phpdbg_print_class_name(pce);
+				} while ((pce = pce->parent));
+			} else {
+				phpdbg_writeln("|-------- User Class %s (not yet linked)", ZSTR_VAL(ce->parent_name));
```

Comment:
i feel the message lacks a bit of clarity (for most of users at least) but do not have better idea right now.

Reply:
Maybe "not yet linked, because declaration has not been encountered yet" or along these lines?

Reply:
But the declaration was encountered, it just wasn't when the class was defined, so it's tricky to be precise.

Reply:
And if we add: "[...] while parsing CHILD_CLASS" ?

Reply:
While declaring child class, but sure that works for me! Maybe a bit long but I rather have it descriptive and long than short.